tests/test_seekrpc.py

```diff
@@ -369,4 +374,149 @@ def test_pclAsyncClient(self):
\ No newline at end of file
+        transport.close()
+
+    def test_pvalueQuery(self):
+        # How this test was set up:
+        # Made a consistent queryFile to generate the random query results, randQueries.txt
+        # Build the pvalue bins from the randQueries, done by seekCreateDB in createSampleDatabase()
+        # Take one query and get the gscores and ranks (take one from random dir, i.e. 3.gscore)
+        # Offline, one time, used the legacy pvalue server to get the score and rank based pvalues for that query
+        # Run the new pvalue server and query against all genes and a partial set of genes
+
+        # Run the queries through the python rpc client
+        from thrift.transport import TTransport, TSocket
+        from thrift.protocol.TBinaryProtocol import TBinaryProtocol
+        socket = TSocket.TSocket('localhost', testPort)
+        transport = TTransport.TBufferedTransport(socket)
+        transport.open()
+        protocol = TBinaryProtocol(transport)
+        client = SeekRPC.Client(protocol)
+
+        # Load the previoulsy-run query results from a pre-selected query.
+        # Will use the 3rd random query 3.*
+        randInputsDir = os.path.join(sampleBcDir, 'randTestInputs')
+        # The results include the list of genes and gene scores.
+        # Read in the gene scores from a seek .gscore binary file:
+        gscoreFile = os.path.join(randInputsDir, '3.gscore')
+        gscores = sutils.readSeekBinaryResultFile(gscoreFile)
+        scorePvalueFile = os.path.join(randInputsDir, '3.score_pvalues')
+        expectedScorePvalues = sutils.readSeekBinaryResultFile(scorePvalueFile)
+        rankPvalueFile = os.path.join(randInputsDir, '3.rank_pvalues')
+        expectedRankPvalues = sutils.readSeekBinaryResultFile(rankPvalueFile)
+
+        # Do the score-based pvalue query for all genes
+        # Note: when doing for all genes no need to provide the gene entrez id list
+        pvalueArgs = SeekRPC.PValueGeneArgs(
+            species = 'sampleBC',
+            # don't specify the geneIDs if getting scores for all genes, in which
+            #  case it is assumed the gscores are in the gene_map order
+            geneScores = gscores,
+            useRank = False
+        )
+        result = client.pvalueGenes(pvalueArgs)
+        assert result.success is True
+        resPvalues = np.array(result.pvalues, dtype=np.float32)
+        # Given 1000 random queries used to build the pvalue tables, accuracy
+        #  should only be to about .001.
+        isEquivalent = np.allclose(resPvalues, expectedScorePvalues, rtol=.05, atol=.001)
+        assert isEquivalent == True
+
+        # Do the rank based pvalue query for all genes
+        pvalueArgs = SeekRPC.PValueGeneArgs(
+            species = 'sampleBC',
+            # don't specify the geneIDs if getting scores for all genes, in which
+            #  case it is assumed the gscores are in the gene_map order
+            geneScores = gscores,
```

Comment:
Add better documentation on this. That if all the gene_scores are passed in then it can calculate the ranks internally and give pvalue for rank.
It was confusing that scores were passed in and useRank=False